### ISSUE TITLE
Trained object detection with yolov8-ghost-p2 cannot successfully convert to tflite correctly

### ISSUE THREAD

**Issue by harufumigithub** [ultralytics/ultralytics#16447]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Export

### Bug

Trained object detection with yolov8-ghost-p2.  
Converted best.pt to tflite with 'yolo export model=best.pt format=tflite int8=true' 
best_int8.tflite is generated but inference with this tflite detect nothing while best.pt (before conversion) detects fine.

### Environment

yolo checks
Ultralytics YOLOv8.2.83 🚀 Python-3.9.12 torch-1.13.1 CUDA:0 (NVIDIA GeForce RTX 3070 Laptop GPU, 7959MiB)
Setup complete ✅ (16 CPUs, 62.4 GB RAM, 131.0/245.0 GB disk)

OS                  Linux-6.8.0-45-generic-x86_64-with-glibc2.35
Environment         Linux
Python              3.9.12
Install             git
RAM                 62.45 GB
CPU                 11th Gen Intel Core(TM) i7-11800H 2.30GHz
CUDA                11.7

numpy               ✅ 1.26.3<2.0.0,>=1.23.0
matplotlib          ✅ 3.8.2>=3.3.0
opencv-python       ✅ 4.9.0.80>=4.6.0
pillow              ✅ 10.2.0>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.27.1>=2.23.0
scipy               ✅ 1.11.4>=1.4.1
torch               ✅ 1.13.1>=1.8.0
torchvision         ✅ 0.14.1>=0.9.0
tqdm                ✅ 4.66.1>=4.64.0
psutil              ✅ 5.9.7
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.1.4>=1.1.4
seaborn             ✅ 0.13.1>=0.11.0
ultralytics-thop    ✅ 2.0.0>=2.0.0

### Minimal Reproducible Example

Do transfer learning with 
model = YOLO('yolov8n-ghost-p2.yaml').load('yolov8n.pt')
results = model.train(data="./detection.yaml", device=0, imgsz=640, epochs=1000, batch=-1)   
Run: yolo export model=best.pt format=tflite int8=true'

Compare the inference result between best.pt and best_int8.tflite:
model(source = ./sample.png, save=True, conf=0.3, save_txt=True, device=0, imgsz=640)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @harufumigithub, thank you for your interest in Ultralytics 🚀!

We recommend visiting the [Docs](https://docs.ultralytics.com) to explore a variety of [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. Many common questions might be answered there!

For this 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to further assist with debugging the issue.

Join the Ultralytics community where it suits you best:
- For real-time chat, head to [Discord](https://ultralytics.com/discord) 🎧
- For in-depth discussions, check out [Discourse](https://community.ultralytics.com)
- Or explore our [Subreddit](https://reddit.com/r/ultralytics) for community knowledge sharing

## Upgrade

To ensure the issue isn't resolved in the latest version, please upgrade to the latest `ultralytics` package:

```bash
pip install -U ultralytics
```

Make sure all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) are installed in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/).

## Environments

YOLOv8 is supported in these up-to-date verified environments:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon**: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, the [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing for all YOLOv8 [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu, checked every 24 hours and on every commit.

This is an automated response. An Ultralytics engineer will assist you soon. 😊

**Comment by Y-T-G** [comment]
You should provide a calibration dataset for int8 conversion by passing the yaml as the `data` argument to `model.export()`.

**Comment by harufumigithub** [comment]
> yolo export model=best.pt format=tflite int8=true'

**Comment by harufumigithub** [comment]
> You should provide a calibration dataset for int8 conversion by passing the yaml as the `data` argument to `model.export()`.

I have 'results = model.train(data="./detection.yaml", device=0, imgsz=640, epochs=1000, batch=-1)' for training.  So should the model.export() include 'data=./detection.yaml'?  I checked best_float16.tflite and best_float32.tflite from ghost_p2 and they worked fine except best_int8.tflite.  So I think conversion for best_int8.tflite has an issue.  I trained with the same dataset with p2 and p6 without ghost option and converted best_int8.tflite usually works fine without the data argument to model.export().

**Comment by glenn-jocher** [comment]
@harufumigithub yes, for INT8 conversion, including `data=./detection.yaml` in `model.export()` is recommended to ensure proper calibration. This helps generate accurate scale values for quantization.

**Comment by harufumigithub** [comment]
> @harufumigithub yes, for INT8 conversion, including `data=./detection.yaml` in `model.export()` is recommended to ensure proper calibration. This helps generate accurate scale values for quantization.

Including 'data=./detection.yaml' seems hang in the last part of conversion process (the last part does not finish after 1 hour):
Ultralytics YOLOv8.2.83 🚀 Python-3.9.12 torch-1.13.1 CPU (11th Gen Intel Core(TM) i7-11800H 2.30GHz)
YOLOv8-ghost-p2 summary (fused): 380 layers, 1,601,704 parameters, 0 gradients, 8.6 GFLOPs

PyTorch: starting from 'runs/detect/train.blob-detr.v25.nucx.ghost-p2/weights/best.pt' with input shape (1, 3, 640, 640) BCHW and output shape(s) (1, 8, 34000) (3.7 MB)

TensorFlow SavedModel: starting export with tensorflow 2.16.2...

ONNX: starting export with onnx 1.15.0 opset 16...
ONNX: slimming with onnxslim 0.1.32...
ONNX: export success ✅ 1.2s, saved as 'runs/detect/train.blob-detr.v25.nucx.ghost-p2/weights/best.onnx' (7.3 MB)
TensorFlow SavedModel: collecting INT8 calibration images from 'data=tag_detection.yaml'
Scanning /mnt/1a8b3160-241c-466f-aa1b-79f0591f7f56/workspace/datasets/tag_detection/val/tiles.640.chma1.chmi3.cache... 2325 images, 0 backgrounds, 0 corrupt: 100%|██████████| 2325/2325 [00:00<?, ?it/s]
TensorFlow SavedModel: starting TFLite export with onnx2tf 1.20.0...

Automatic generation of each OP name started ========================================
Automatic generation of each OP name complete!

Model loaded ========================================================================

Model conversion started ============================================================
saved_model output started ==========================================================
saved_model output complete!
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
W0000 00:00:1727183118.656295    4236 tf_tfl_flatbuffer_helpers.cc:390] Ignored output_format.
W0000 00:00:1727183118.656321    4236 tf_tfl_flatbuffer_helpers.cc:393] Ignored drop_control_dependency.
Float32 tflite output complete!
W0000 00:00:1727183119.435429    4236 tf_tfl_flatbuffer_helpers.cc:390] Ignored output_format.
W0000 00:00:1727183119.435449    4236 tf_tfl_flatbuffer_helpers.cc:393] Ignored drop_control_dependency.
Float16 tflite output complete!
Input signature information for quantization
signature_name: serving_default
input_name.0: images shape: (1, 640, 640, 3) dtype: <dtype: 'float32'>
W0000 00:00:1727183121.356893    4236 tf_tfl_flatbuffer_helpers.cc:390] Ignored output_format.
W0000 00:00:1727183121.356916    4236 tf_tfl_flatbuffer_helpers.cc:393] Ignored drop_control_dependency.
Dynamic Range Quantization tflite output complete!
W0000 00:00:1727183125.320798    4236 tf_tfl_flatbuffer_helpers.cc:390] Ignored output_format.
W0000 00:00:1727183125.320819    4236 tf_tfl_flatbuffer_helpers.cc:393] Ignored drop_control_dependency.
fully_quantize: 0, inference_type: 6, input_inference_type: FLOAT32, output_inference_type: FLOAT32
INT8 Quantization tflite output complete!
W0000 00:00:1727185340.800359    4236 tf_tfl_flatbuffer_helpers.cc:390] Ignored output_format.
W0000 00:00:1727185340.800377    4236 tf_tfl_flatbuffer_helpers.cc:393] Ignored drop_control_dependency.

**Comment by Y-T-G** [comment]
Cancel and try it again.

**Comment by harufumigithub** [comment]
> Cancel and try it again.

It gets stuck at the same place, and the process never finishes, even after one hour.

**Comment by glenn-jocher** [comment]
It seems like the issue might be related to the calibration process. Try reducing the batch size or image size to see if that resolves the problem. If it persists, consider checking your environment for any resource constraints.

**Comment by harufumigithub** [comment]
If I don't specify data=./detection.yaml, conversion can complete conversion to best_int8.tflite.  
Inference results:
best.pt (inference works)
best_float16.tflite  (inference works)
best_float32.tflite (inference works)
best_int8.tflite (inference does NOT work)
In the meanwhile, I will retrain with smaller batch size.

**Comment by glenn-jocher** [comment]
It seems the issue might be specific to the ghost model. You could try using a different calibration dataset or adjusting the calibration parameters. If the problem persists, consider reaching out to the community for further insights.
